From kristen.c.accardi@intel.com  Tue Dec 19 22:34:09 2006
Date: Sat, 16 Dec 2006 15:26:04 -0800
From: Peter Chubb <peterc@gelato.unsw.edu.au>
From: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
To: gregkh@suse.de
Cc: Peter Chubb <peterc@gelato.unsw.edu.au>, Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Subject: acpiphp: Link-time error for PCI Hotplug
Message-Id: <20061216152604.84153541.kristen.c.accardi@intel.com>
Mime-Version: 1.0
Content-Type: text/plain; charset=US-ASCII

	I'm seeing:
 `acpiphp_glue_exit' referenced in section `.init.text' of
drivers/built-in.o: defined in discarded section `.exit.text' of
drivers/built-in.o

when trying to compile an IA64 kernel with PCI hotplug enabled.

I suggest this patch:

Signed-off-by: Peter Chubb <peterc@gelato.unsw.edu.au>
Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/acpiphp_glue.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/acpiphp_glue.c
+++ gregkh-2.6/drivers/pci/hotplug/acpiphp_glue.c
@@ -1682,7 +1682,7 @@ int __init acpiphp_glue_init(void)
  *
  * This function frees all data allocated in acpiphp_glue_init()
  */
-void __exit acpiphp_glue_exit(void)
+void  acpiphp_glue_exit(void)
 {
 	acpi_pci_unregister_driver(&acpi_pci_hp_driver);
 }
